File: crates/tldr-cli/tests/fixtures/vuln_migration_v1/csharp/command_injection_string_literal_fp.cs
// vt=CommandInjection lang=csharp — names below are inside strings/comments only
// Request.Query[...] is a source. cmd.ExecuteReader, Response.Write,
// System.Diagnostics.Process.Start, System.IO.File.Open, JavaScriptSerializer are sinks.

public class DocsOnly {
    public string Docs() {
        string doc = "Request.Query[id] flows into cmd.ExecuteReader(SELECT ... )";
        string more = "Response.Write, Process.Start, System.IO.File.Open, JavaScriptSerializer";
        return doc + more;
    }
}

File: crates/tldr-cli/tests/fixtures/vuln_migration_v1/typescript/command_injection_string_literal_fp.ts
// vt=CommandInjection lang=typescript — names below are inside strings/comments only
// Documentation: req.query.id is a source. db.query(...) is a sink.
// child_process.exec, fs.readFileSync, fetch, eval, res.send are sinks.

export function docsOnly() {
    const doc: string = "req.query.id flows into db.query(SELECT ...) and child_process.exec";
    const more: string = "fetch(url) and fs.readFileSync(p) and eval(code) and res.send(html)";
    return doc + more;
}
